modules/broadcast/src/backend/api.ts

```diff
@@ -26,35 +28,37 @@ export default async (bp: SDK, db: Database) => {
           filteringConditions: row.filters && JSON.parse(row.filters)
         }
       })
-
       res.send(broadcasts)
     })
-  })
+  )
 
-  router.put('/', (req, res, next) => {
-    const { date, time, timezone, content, type, filters } = req.body
-    const { botId } = req.params
+  router.put(
+    '/',
+    asyncMiddleware(async (req, res) => {
+      const { date, time, timezone, content, type, filters } = req.body
+      const { botId } = req.params
 
-    db.addSchedule({ botId, date, time, timezone, content, type, filters }).then(id => res.send({ id: id }))
-  })
+      const { id } = await db.addSchedule({ botId, date, time, timezone, content, type, filters })
+      res.send({ id })
+    })
+  )
 
-  router.post('/', (req, res, next) => {
-    const { id, date, time, timezone, content, type, filters } = req.body
+  router.post(
```

Comment:
wow those 2 routes are mixed up, post is usally used to create and put to update hehe. but in our case we should wither combine the 2 routes or make sure both update and create are post

Reply:
Yup, i'll change them. I didn't even realize that, just added the async middleware haha